### PR TITLE
feat(smart-contracts): script to automate gnosis setup

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This PR adds hardhat tasks to automate the setup and use of gnosis safes. Here is what it should achieve:

- [x] create a new safe with a custom list of owners
- [x] duplicate a safe (create a new safe using owners and threshold from another network)
- [x] submit a lock template upgrade 
- [x] submit an unlock upgrade
- [x] more generally, parse and submit any possible tx from a json of abi and params
- [x] works on a fork of mainnet

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

Run the tests 

```
RUN_MAINNET_FORK=1 yarn hardhat test test/scripts/gnosis.mainnet.js
```

### CODE REVIEW

**Review by julien51 — APPROVED** [review]
Ready

File: smart-contracts/scripts/multisig/create.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
 const { ethers } = require('hardhat')
 const { SafeFactory } = require('@gnosis.pm/safe-core-sdk')
 const EthersAdapter = require('@gnosis.pm/safe-ethers-lib').default
+const getOwners = require('./owners')
 
-async function main({ owners, threshold }) {
+const UNLOCK_MAINNET_MULTISIG_ADDRESS =
```
Comment (julien51) [inline smart-contracts/scripts/multisig/create.js:6]: Let's get it from the network package please? (and if it is not there, just add it there for all networks?)

Comment (clemsos) [reply smart-contracts/scripts/multisig/create.js:6]: the other way is to get it with `unlock.owner()` which is maybe safer and prevents overloading  the networks package

File: smart-contracts/scripts/multisig/_helpers.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+  }
+
+  //get unlock contract
+  const { unlockAddress } = networks[chainId]
+  const { interface } = await ethers.getContractFactory('Unlock')
+  const unlock = new ethers.Contract(unlockAddress, interface, provider)
+
+  // get address
+  const safeAddress = await unlock.owner()
```
Comment (julien51) [inline smart-contracts/scripts/multisig/_helpers.js:35]: Unfortunately as discussed it is better to use hardcoded value sin the `networks` package

**Comment by clemsos** [comment]
can rebase once #9073 is merged

**Comment by clemsos** [comment]
also need to rebase after #9094 is merged

**Review by julien51 — APPROVED** [review]
LGTM!

File: smart-contracts/test/helpers/multisig.js [hunk 3]
```diff
@@ -1,34 +1,16 @@
...
-const encodeUpgradeTxData = async ({ proxyAddress, implementation }) => {
-  // build upgrade tx
-  const proxy = await ethers.getContractAt(proxyABI, proxyAddress)
-  const data = proxy.interface.encodeFunctionData('upgrade', [
-    proxyAddress,
-    implementation,
-  ])
-  return data
-}
+const MULTISIG_ADDRESS_OWNER = '0xDD8e2548da5A992A63aE5520C6bC92c37a2Bcc44'
```
[the remark this answers is not in the document]
Comment (julien51) [reply smart-contracts/test/helpers/multisig.js:6]:
```suggestion
const MULTISIG_ADDRESS_OWNER = '0xF5C28ce24Acf47849988f147d5C75787c0103534'
```

**Pull request merged**